docs/e2e-test-guide.md

```diff
@@ -0,0 +1,164 @@
+# Argo CD Operator E2E Test Guide
+
+E2E tests are written using [KUTTL](https://kuttl.dev/docs/#install-kuttl-cli).
+
+## Requirements
+
+This test suite assumes that an Argo CD Operator is installed on the cluster or running locally using `make run`.
```

Comment:
The cluster-config tests should depend on the `ARGOCD_CLUSTER_CONFIG_NAMESPACES` env var being set, but I don't see it being set anywhere.